drivers/net/ppp.c

```diff
@@ -168,6 +173,61 @@ static void ppp_change_state(struct ppp_driver_context *ctx,
 	ctx->state = new_state;
 }
 
+static int ppp_send_flush(struct ppp_driver_context *ppp, int off)
+{
+	if (!IS_ENABLED(CONFIG_NET_TEST)) {
+		uart_pipe_send(ppp->send_buf, off);
+	}
+
+	return 0;
+}
+
+static int ppp_send_bytes(struct ppp_driver_context *ppp,
+			  const u8_t *data, int len, int off)
+{
+	int i;
+
+	for (i = 0; i < len; i++) {
+		ppp->send_buf[off++] = data[i];
+
+		if (off >= sizeof(ppp->send_buf)) {
+			off = ppp_send_flush(ppp, off);
+		}
+	}
+
+	return off;
+}
+
+#if defined(CONFIG_PPP_CLIENT_CLIENTSERVER)
+static void ppp_handle_client(struct ppp_driver_context *ppp, u8_t byte)
+{
+	static const char *client = "CLIENT";
+	static const char *clientserver = "CLIENTSERVER";
+	int offset;
+
+	if (ppp->client_index >= strlen(client)) {
```

Comment:
Please remove the `strlen()` calls by by `sizeof()`, so something like this
```
#define CLIENT "CLIENT"
#define CLIENTSERVER "CLIENTSERVER"

	static const char *client = CLIENT;
	static const char *clientserver = CLIENTSERVER;
...
	if (ppp->client_index >= (sizeof(CLIENT) - 1)) {
```
etc. This way we avoid multiple runtime calls to `strlen()` as the `sizeof()` is done at compile time.

Reply:
done and squashed